src/Config/OssIndexServerConfig.ts

```diff
@@ -45,6 +46,14 @@ export class OssIndexServerConfig extends Config {
     return this.cacheLocation;
   }
 
+  public async clearCache(): Promise<void> {
+    if (this.cacheLocation) {
+      await storage.init({ dir: this.cacheLocation });
+      return await storage.clear();
+    }
+    throw new Error('There was an error clearing the Cache, did you remember to set the location in config?')
```

Comment:
Not sure if this error message is meaningful, and it's not being hit when i delete the cache itself:

```
ajurgenson@ArtieSonaDell:~/git_repos/auditjs$ npm run start -- ossi --clear

> auditjs@4.0.1-alpha.29 start /home/ajurgenson/git_repos/auditjs
> node ./bin/index.js "ossi" "--clear"

internal/fs/utils.js:220
    throw err;
    ^

Error: ENOENT: no such file or directory, open '/home/ajurgenson/.ossindex/.oss-index-config'
    at Object.openSync (fs.js:440:3)
    at Object.readFileSync (fs.js:342:35)
    at OssIndexServerConfig.getConfigFromFile (/home/ajurgenson/git_repos/auditjs/bin/Config/OssIndexServerConfig.js:65:45)
    at Object.<anonymous> (/home/ajurgenson/git_repos/auditjs/bin/index.js:161:16)
    at Module._compile (internal/modules/cjs/loader.js:955:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)
    at Module.load (internal/modules/cjs/loader.js:811:32)
    at Function.Module._load (internal/modules/cjs/loader.js:723:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10)
    at internal/main/run_main_module.js:17:11 {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/ajurgenson/.ossindex/.oss-index-config'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! auditjs@4.0.1-alpha.29 start: `node ./bin/index.js "ossi" "--clear"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the auditjs@4.0.1-alpha.29 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ajurgenson/.npm/_logs/2020-02-11T22_29_00_724Z-debug.log
```

There is a public exists function in Config.ts, you hsould use that to check if the config actually exists

Reply:
@ButterB0wl https://github.com/sonatype-nexus-community/auditjs/pull/153/commits/f442b7ad87b966e7051bbf46f14853c8ea90cfd7 Is this what you had in mind?

Reply:
@DarthHater @ButterB0wl Another way I can try doing this is

```  public async clearCache(): Promise<void> {
    if (this.exists()) {
      await storage.init({ dir: this.cacheLocation });
      try {
        await storage.clear();
      }
      catch {
        throw new Error('There was an error clearing the Cache, did you remember to set the location in config? Clear will only clear the location set in config')
      }
    }
    throw new Error('There was an error clearing the Cache, did you remember to set the location in config? Clear will only clear the location set in config')
  }```